packages/timeline-state-resolver/src/integrations/vmix/vMixResponseStreamReader.ts

```diff
@@ -0,0 +1,104 @@
+import { EventEmitter } from 'eventemitter3'
+
+type ResponseStreamReaderEvents = {
+	response: [response: Response]
+	error: [error: Error]
+}
+
+export interface Response {
+	command: string
+	response: 'OK' | 'ER'
+	message: string
+	body?: string
+}
+
+const RESPONSE_REGEX = /^(?<command>\w+)\s+(?<response>OK|ER|\d+)(\s+(?<responseMsg>.*))?/i
+
+/**
+ * A receiver for vMix responses
+ */
+export class VMixResponseStreamReader extends EventEmitter<ResponseStreamReaderEvents> {
+	private _unprocessedLines: string[] = []
+	private _lineRemainder = ''
+
+	reset() {
+		this._unprocessedLines = []
+		this._lineRemainder = ''
+	}
+
+	processIncomingData(data: Buffer) {
+		const string = this._lineRemainder + data.toString('utf-8')
+		this._lineRemainder = ''
+		const lines = string.split('\r\n')
+
+		if (lines[lines.length - 1] === '') {
+			// the data ends with a newline
+			this._unprocessedLines.push(...lines)
+		} else {
+			const incompleteLine = lines.pop()
+			this._unprocessedLines.push(...lines)
+			// we need to keep the remaining incomplete line
+			if (incompleteLine != null) {
```

Comment:
```suggestion
			if (incompleteLine) {
```
`!= null` is an odd choice here, as im pretty sure `pop()` returns `string | undefined`. May as well be a falsey check though

Also I think sonarcloud will complain about usage of `!=` and `==`. So they are better to avoid without a good reason for them

Reply:
I'm personally a huge fan of `== null` and `!= null` as a way of checking whether a value is/isn't nullish. If eslint has a rule that allows it, wouldn't sonarcloud too? https://eslint.org/docs/latest/rules/eqeqeq#smart

Reply:
I didn't realise that these checks had smart and useful behaviour. I assume our eslint must be configured to allow these ones otherwise it would have complained, so maybe sonar will too. 
For now lets leave them as is, I've added it to our internal devmeet agenda, in case anyone feels strongly against this.